components/script/dom/messagechannel.rs

```diff
@@ -0,0 +1,59 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use dom::bindings::codegen::Bindings::MessageChannelBinding::{MessageChannelMethods, Wrap};
+use dom::bindings::error::{Error, Fallible};
+use dom::bindings::js::{JS, Root};
+use dom::bindings::reflector::{Reflector, reflect_dom_object};
+use dom::globalscope::GlobalScope;
+use dom::messageport::MessagePort;
+use dom_struct::dom_struct;
+
+#[dom_struct]
+pub struct MessageChannel {
+    reflector_: Reflector,
+    port1: JS<MessagePort>,
+    port2: JS<MessagePort>,
+}
+
+impl MessageChannel {
+    // https://html.spec.whatwg.org/multipage/#dom-messagechannel
+    pub fn Constructor(global: &GlobalScope) -> Fallible<Root<MessageChannel>> {
+        let incumbent = match GlobalScope::incumbent() {
```

Comment:
I tried that, it said something like `convertFrom is not implemented for Result<_, dom::bindings::Error>`